src/sentry/integrations/slack/service.py

```diff
@@ -0,0 +1,206 @@
+
+from sentry.integrations.repository import get_default_issue_alert_repository
+from sentry.integrations.repository.issue_alert import (
+    IssueAlertNotificationMessage,
+    IssueAlertNotificationMessageRepository,
+)
+from sentry.integrations.slack import BlockSlackMessageBuilder, SlackClient
+from sentry.models.activity import Activity
+from sentry.models.organization import OrganizationStatus
+from sentry.models.rule import Rule
+from sentry.notifications.notifications.activity import EMAIL_CLASSES_BY_TYPE
+from sentry.services.hybrid_cloud.integration import integration_service
+
+_default_logger = getLogger(__name__)
+
+
+class RuleDataError(Exception):
+    pass
+
+
+class SlackService:
+    def __init__(
+        self,
+        notification_message_repository: IssueAlertNotificationMessageRepository,
+        message_block_builder: BlockSlackMessageBuilder,
+        logger: Logger,
+    ) -> None:
+        self._notification_message_repository = notification_message_repository
+        self._slack_block_builder = message_block_builder
+        self._logger = logger
+
+    @classmethod
+    def default(cls) -> SlackService:
+        return SlackService(
+            notification_message_repository=get_default_issue_alert_repository(),
+            message_block_builder=BlockSlackMessageBuilder(),
+            logger=_default_logger,
+        )
+
+    def notify_all_threads_for_activity(self, activity: Activity) -> None:
+        """
+        For an activity related to an issue group, send notifications in a Slack thread to all parent notifications for
+        that specific group and project.
+
+        If the group is not associated with an activity, return early as there's nothing to do.
+        If the user is not associated with an activity, return early as we only care about user activities.
+        """
+        if activity.group is None:
+            self._logger.info(
+                "no group associated on the activity, nothing to do",
+                extra={
+                    "activity_id": activity.id,
+                },
+            )
+            return None
+
+        if activity.user_id is None:
+            self._logger.info(
+                "machine/system updates are ignored at this time, nothing to do",
+                extra={
+                    "activity_id": activity.id,
+                },
+            )
+            return None
+
+        # The same message is sent to all the threads, so this needs to only happen once
+        notification_to_send = self._get_notification_message_to_send(activity=activity)
+        if not notification_to_send:
+            self._logger.info(
+                "notification to send is invalid",
+                extra={
+                    "activity_id": activity.id,
+                },
+            )
+            return None
+
+        try:
+            integration = integration_service.get_integration(
+                organization_id=activity.project.organization_id,
```

Comment:
What I mean is that not finding integration doesn't have anything to do with activity but rather organization. So the code would look like this:

```
organization_id = activity.project.organization_id
# can organization be None? Check here instead of integration related code
integration = getIntegrationForOrganization() # this method has the try catch inside it
```

Reply:
Yeah that makes sense! I can def add that helper function